include/albatross/src/core/distribution.hpp

```diff
@@ -100,7 +100,28 @@ struct MarginalDistribution : public DistributionBase<MarginalDistribution> {
   }
 
   bool operator==(const MarginalDistribution &other) const {
-    return (mean == other.mean && covariance == other.covariance);
+    const double epsilon = 0.1;
+
+    if (mean.size() != other.mean.size()) {
+      return false;
+    }
+    for (int i = 0; i < mean.size(); i++) {
+      if (fabs(mean[i] - other.mean[i]) > epsilon) {
+        return false;
+      }
+    }
+
+    if (covariance.diagonal().size() != other.covariance.diagonal().size()) {
+      return false;
+    }
+    for (int i = 0; i < covariance.diagonal().size(); i++) {
+      if (fabs(covariance.diagonal()[i] - other.covariance.diagonal()[i]) >
+          epsilon) {
+        return false;
+      }
+    }
+
+    return true;
```

Comment:
I'm not really sure what this will do with different sized distributions ... but you could take advantage of Eigen's helpers:
```suggestion
    const bool mean_approx_equal = mean.isApprox(other.mean, epsilon);
    const bool cov_approx_equal = covariance.diagonal().isApprox(other.covariance.diagonal(), epsilon);
    return mean_approx_equal && cov_approx_equal;
```